components/widgets/team_icon/team_icon.tsx

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {imageURLForTeam} from 'utils/utils.jsx';
+
+import './team_icon.scss';
+
+type Props = {
+
+    /** Team icon URL (when available) */
+    url?: string;
+
+    /** Team display name (used for the initials) if icon URL is not set */
+    name: string;
+
+    /**
+     * Size of the icon, "small", "regular" or "large".
+     *
+     * @default "regular"
+     **/
+    size?: 'small'|'regular'|'large';
```

Comment:
We were using for the avatar the sizes `xs|sm|md|lg|xl|xxl` I think we should be consistent with the pattern, so we should change it here, or in the Avatar component. Not necessarily to have the same number of sizes, but use the same names, in this case I would use `sm|md|lg`.